packages/config/src/handlers/locations/locationTreeSolver.ts

```diff
@@ -19,9 +19,8 @@ import { fetchFromHearth } from '@config/services/hearth'
 import { MongoClient } from 'mongodb'
 import { env } from '@config/environment'
 
-const client = new MongoClient(env.HEARTH_MONGO_URL)
-
 export const resolveLocationChildren = async (id: UUID) => {
+  const client = new MongoClient(HEARTH_MONGO_URL)
```

Comment:
I was reading through this article: https://www.mongodb.com/community/forums/t/using-node-driver-v5-when-is-mongoclient-connect-required/212008/7

So perhaps would it not be better to define the client in [this](https://github.com/opencrvs/opencrvs-core/blob/develop/packages/config/src/config/database.ts) file? That way we can close the connection when the server is stopped. Although I'm pretty sure we are not calling the `stop` function anywhere but I think it's still better to have only one connection per server instead per function call

Reply:
Yeah, I did come across that as a potentially preferable option but I just copied the way it was done in other services. But I think the connection pooling is quite a big deal so a single instance looks like a better way to go.
For consistency, I can change all the services to work the same way.

Reply:
Done for all services.